go/vt/vttablet/tabletserver/tx_pool.go

```diff
@@ -182,6 +182,20 @@ func (axp *TxPool) Begin(ctx context.Context, useFoundRows bool) (int64, error)
 		}
 		return 0, err
 	}
+
+	switch txIsolation {
+	case querypb.ExecuteOptions_DEFAULT:
+		break
+	case querypb.ExecuteOptions_REPEATABLE_READ, querypb.ExecuteOptions_READ_COMMITTED, querypb.ExecuteOptions_READ_UNCOMMITTED, querypb.ExecuteOptions_SERIALIZABLE:
+		query := fmt.Sprintf("set transaction isolation level %v", strings.Replace(txIsolation.String(), "_", " ", 1))
```

Comment:
I think it's better to use a `map[querypb.ExecuteOptions]string` here. We could even map it to the full statement, thereby avoiding the Sprintf.